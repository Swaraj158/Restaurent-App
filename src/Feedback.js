import bootstrap from 'bootstrap/dist/css/bootstrap.min.css'
import { useState } from 'react'

export default function FeedBack(){

    const [username,setusername]=useState('')
    const [contact,setcontact]=useState('')
    const [feedback,setfeedback]=useState('')
    const [allfeedbacks,setallFeedBacks]=useState([])

    function submitFeedBack(event){
        event.preventDefault();

        var all = [...allfeedbacks]
        var currFeed = {name:username,contact:contact,feedback:feedback}
        all.push(currFeed)
        setallFeedBacks(all)

    }

    function reset(){
        setusername('')
        setcontact('')
        setfeedback('')
    }

    const feed = allfeedbacks.map((fed,index)=>{
        return(
        <tr>
            <td>{index+1}</td>
            <td>{fed.name}</td>
            <td>{fed.contact}</td>
            <td>{fed.feedback}</td>
        </tr>
    )})

    return(
        <div className="container">
            <div className="row justify-content-center">
                <div className="col-md-6 ">
                    <h2 style={{color:"yellow",margin:"20px"}}>FeedBack</h2>    
                    <input type="text" data-testid="user-input" placeholder="Name" className='form-control' value={username} onChange={(e)=>{setusername(e.target.value)}}/>
                    <input type="text" placeholder="Contact" className='form-control' value={contact} onChange={(e)=>{setcontact(e.target.value)}}/>
                    <input type="textarea" placeholder="FeedBack" className='form-control' value={feedback} onChange={(e)=>{setfeedback(e.target.value)}}/>
            
                    <button className="btn btn-lg btn-primary" data-testid="input-submit" onClick={submitFeedBack}>Submit</button>
                    <button className="btn btn-lg btn-neutral" data-testid="input-reset" Click={reset}>Reset</button>
                </div>
            
                <div className="col-md-10">
                    <div className="order-form">
                        <h1 style={{margin:"30px"}}>Past FeedBacks</h1>
                        <table className="table table-dark table-striped table-bordered">
                            <thead className="thead">
                                <tr>
                                    <th>Sr.no.</th>
                                    <th>Name</th>
                                    <th>Contact</th>
                                    <th>FeedBack</th>
                                </tr>
                            </thead>
                            <tbody>
                                {feed}
                            </tbody>
                        </table>
                    </div>
                </div>
            </div>
        </div>
    )
}