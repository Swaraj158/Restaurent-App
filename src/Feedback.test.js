import React from "react";
import { render,fireEvent, getByTestId } from "@testing-library/react";
import Feedback from './Feedback';
import '@testing-library/jest-dom/extend-expect'


describe('Test Feedback', () => {

    it("renders properly",()=>{
      const {baseElement} = render(<Feedback/>);
      expect(baseElement).toBeInTheDocument();
    })

    it("matches snapshot",()=>{
      const {baseElement}=render(<Feedback/>)
      expect(baseElement).toMatchSnapshot();
    })

    it("Receives Input text",()=>{
        const testuser = "testuser";
        const {getByTestId} = render(<Feedback/>);
        const input1 = getByTestId("user-input");
        expect(input1).toBeInTheDocument();
       
       
        fireEvent.change(input1,{target:{value:testuser}});
        expect(input1).toHaveValue(testuser);
                
        const buttonSubmit = getByTestId("input-submit");
        expect(buttonSubmit).toBeInTheDocument();

        const buttonReset = getByTestId("input-reset");
        expect(buttonReset).toBeInTheDocument();
    })

});
