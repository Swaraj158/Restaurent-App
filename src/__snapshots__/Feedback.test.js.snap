// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Feedback matches snapshot 1`] = `
<body>
  <div>
    <div
      class="container"
    >
      <div
        class="row justify-content-center"
      >
        <div
          class="col-md-6 "
        >
          <h2
            style="color: yellow; margin: 20px;"
          >
            FeedBack
          </h2>
          <input
            class="form-control"
            data-testid="user-input"
            placeholder="Name"
            type="text"
            value=""
          />
          <input
            class="form-control"
            placeholder="Contact"
            type="text"
            value=""
          />
          <input
            class="form-control"
            placeholder="FeedBack"
            type="textarea"
            value=""
          />
          <button
            class="btn btn-lg btn-primary"
            data-testid="input-submit"
          >
            Submit
          </button>
          <button
            class="btn btn-lg btn-neutral"
            data-testid="input-reset"
          >
            Reset
          </button>
        </div>
        <div
          class="col-md-10"
        >
          <div
            class="order-form"
          >
            <h1
              style="margin: 30px;"
            >
              Past FeedBacks
            </h1>
            <table
              class="table table-dark table-striped table-bordered"
            >
              <thead
                class="thead"
              >
                <tr>
                  <th>
                    Sr.no.
                  </th>
                  <th>
                    Name
                  </th>
                  <th>
                    Contact
                  </th>
                  <th>
                    FeedBack
                  </th>
                </tr>
              </thead>
              <tbody />
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
